examples/plot_simulate_somato.py

```diff
@@ -94,45 +146,96 @@
-    'evprox1', mu=20.0, sigma=0., numspikes=1, weights_ampa=weights_ampa_p,
-    weights_nmda=weights_nmda_p, location='proximal',
-    synaptic_delays=synaptic_delays_p, seedcore=6)
+    'evdist1', mu=32., sigma=2.5, numspikes=1, sync_within_trial=True,
+    weights_ampa=weights_ampa_d, weights_nmda=weights_nmda_d,
+    location='distal', synaptic_delays=synaptic_delays_d, seedcore=6)
+
+# Late distal drive
+weights_ampa_d = {'L2_basket': 0.0041, 'L2_pyramidal': 0.0019,
+                  'L5_pyramidal': 0.0018}
+weights_nmda_d = {'L2_basket': 0.0032, 'L2_pyramidal': 0.0018,
+                  'L5_pyramidal': 0.0017}
+synaptic_delays_d = {'L2_basket': 0.1, 'L2_pyramidal': 0.1,
+                     'L5_pyramidal': 0.1}
 
-dpl = simulate_dipole(net, n_trials=1)
+net.add_evoked_drive(
+    'evdist2', mu=84., sigma=4.5, numspikes=1, sync_within_trial=True,
+    weights_ampa=weights_ampa_d, weights_nmda=weights_nmda_d,
+    location='distal', synaptic_delays=synaptic_delays_d, seedcore=2)
 
-trial_idx = 0
-import matplotlib.pyplot as plt
-fig, axes = plt.subplots(2, 1, sharex=True, figsize=(6, 6))
-dpl[trial_idx].plot(ax=axes[0], show=False)
-net.cell_response.plot_spikes_hist(ax=axes[1])
-net.cell_response.plot_spikes_raster()
+###############################################################################
+# Now we run the simulation over 2 trials so that we can plot the average
+# aggregate dipole. For a better match to the empirical waveform, set
+# ``n_trials`` to be >=25.
+n_trials = 2
+# n_trials = 25
+with JoblibBackend(n_jobs=2):
+    dpls = simulate_dipole(net, n_trials=n_trials)
+
+###############################################################################
+# Finally, we plot the driving spike histogram, empirical and simulated median
+# nerve evoked response waveforms, and output spike histogram.
+fig, axes = plt.subplots(3, 1, sharex=True, figsize=(6, 6))
+net.cell_response.plot_spikes_hist(ax=axes[0],
+                                   spike_types=['evprox', 'evdist'],
+                                   show=False)
+axes[1].axhline(0, c='k', ls=':', label='_nolegend_')
+axes[1].plot(1e3 * stc.times, dipole_tc, 'r--')
+average_dipoles(dpls).plot(ax=axes[1], show=False)
+axes[1].legend(['MNE label average', 'HNN simulation'])
+axes[1].set_ylabel('Current Dipole (nAm)')
+net.cell_response.plot_spikes_raster(ax=axes[2])
 
 ###############################################################################
 # .. LINKS
 #
 # .. _MNE: https://mne.tools/
+# .. _this MNE-python example: https://mne.tools/stable/auto_examples/inverse/plot_label_source_activations.html#sphx-glr-auto-examples-inverse-plot-label-source-activations-py # noqa
+# .. |mne_label_fig| image:: https://user-images.githubusercontent.com/20212206/106524603-cfe75c80-64b0-11eb-9607-3415195c3e7a.png # noqa
+#   :width: 400
+# .. |mne_activity_fig| image:: https://user-images.githubusercontent.com/20212206/106524542-b514e800-64b0-11eb-835e-497454e75eb9.png # noqa
```

Comment:
you do this as you don't want to have to setup pyvista + vtk on your CI?

Reply:
yes it's not really worth the effort to set up headless 3D rendering at this stage of HNN development cycle. We might do it after the 0.1 release though (which will happen soon)

Reply:
Out of curiosity, what is the primary difficulty with setting up the 3D rendering packages on CI?

Reply:
OpenGL requires hardware (GPU) which the headless servers don't have/expose. A software emulation layer is needed, which I'm guessing for something as complex as VTK can be a bit of a hassle. Could also be a performance-issue. It's certainly doable (look at mne!), but I'm glad we decided to side-step it for now: there's plenty to work on as it is :)